File: app/src/main/java/com/example/administrator/androidtest/demo/BizDemo/TikTokDemo/Detail/VideoActionComponent.java
package com.example.administrator.androidtest.demo.BizDemo.TikTokDemo.Detail;

import android.widget.ImageView;
import android.widget.TextView;

import com.bear.libcomponent.component.ViewComponent;
import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.bitmap.CircleCrop;
import com.example.administrator.androidtest.databinding.ItemTiktokVideoDetailBinding;
import com.example.administrator.androidtest.demo.BizDemo.TikTokDemo.TiktokVideoDetailInfo;

public class VideoActionComponent extends ViewComponent<ItemTiktokVideoDetailBinding> {
    private ImageView authorAvatarIv;
    private ImageView musicAvatarIv;
    private TextView likeCountTv;
    private TextView commentCountTv;
    private TextView collectCountTv;
    private TextView shareCountTv;

    public VideoActionComponent(ItemTiktokVideoDetailBinding binding) {
        super(binding);
    }

    @Override
    protected void onCreate() {
        super.onCreate();
    }

    @Override
    protected void onAttachViewBinding(ItemTiktokVideoDetailBinding binding) {
        super.onAttachViewBinding(binding);
        authorAvatarIv = getBinding().authorAvatarIv;
        musicAvatarIv = getBinding().musicAvatarIv;
        likeCountTv = getBinding().likeCountTv;
        commentCountTv = getBinding().commentCountTv;
        collectCountTv = getBinding().collectCountTv;
        shareCountTv = getBinding().shareCountTv;
    }

    private void bindVideoDetailInfo(TiktokVideoDetailInfo videoDetailInfo) {
        Glide.with(getContext()).load(videoDetailInfo.authorImgUrl).transform(new CircleCrop()).into(authorAvatarIv);
        Glide.with(getContext()).load(videoDetailInfo.musicImgUrl).transform(new CircleCrop()).into(musicAvatarIv);
        likeCountTv.setText(String.format("%d", videoDetailInfo.likeCount));
        commentCountTv.setText(String.format("%d", videoDetailInfo.likeCount));
        collectCountTv.setText(String.format("%d", videoDetailInfo.likeCount));
        shareCountTv.setText(String.format("%d", videoDetailInfo.likeCount));
    }

    public void bindVideoDetailInfoExt(TiktokVideoDetailInfo videoDetailInfo) {
        bindVideoDetailInfo(videoDetailInfo);
    }
}

File: app/src/main/java/com/example/administrator/androidtest/demo/bizdemo/tiktokdemo/detail/VideoActionComponent.kt
package com.example.administrator.androidtest.demo.bizdemo.tiktokdemo.detail

import com.bear.libcommon.util.ToastUtil
import com.bear.libcomponent.component.ui.ViewComponent
import com.bear.libcomponent.core.IComponent
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.bitmap.CircleCrop
import com.example.administrator.androidtest.databinding.ComponentTiktokVideoActionBinding
import com.example.administrator.androidtest.demo.bizdemo.tiktokdemo.TiktokDetailInfo

class VideoActionComponent(binding: ComponentTiktokVideoActionBinding) : ViewComponent<ComponentTiktokVideoActionBinding>(binding),
    IVideoActionComponent {
    override fun onCreate() {
        super.onCreate()
        requireBinding().apply {
            authorAvatarIv.setOnClickListener {
                ToastUtil.showToast("点击了作者头像")
            }
        }
    }

    private fun _bindVideoDetailInfo(videoDetailInfo: TiktokDetailInfo) {
        requireBinding().apply {
            Glide.with(requireContext()).load(videoDetailInfo.authorImgUrl).transform(CircleCrop()).into(authorAvatarIv)
            Glide.with(requireContext()).load(videoDetailInfo.musicImgUrl).transform(CircleCrop()).into(musicAvatarIv)
            likeCountTv.text = String.format("%d", videoDetailInfo.likeCount)
            commentCountTv.text = String.format("%d", videoDetailInfo.likeCount)
            collectCountTv.text = String.format("%d", videoDetailInfo.likeCount)
            shareCountTv.text = String.format("%d", videoDetailInfo.likeCount)
        }
    }

    override fun bindVideoDetailInfo(videoDetailInfo: TiktokDetailInfo?) {
        videoDetailInfo ?: return
        _bindVideoDetailInfo(videoDetailInfo)
    }
}

interface IVideoActionComponent : IComponent {
    fun bindVideoDetailInfo(videoDetailInfo: TiktokDetailInfo?)
}
